subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -187,13 +187,24 @@ static int generate_prand(uint32_t *dest)
 		}
 
 		/* Validate Prand: Must contain both a 1 and a 0 */
-		if (*dest != 0 && *dest != 0x3FFFFF) {
-			valid = true;
+		for (uint8_t i = 0; i < BT_CSIP_SIH_PRAND_SIZE; i++) {
+			if (dest[i]) {
+				has_ones = true;
+			}
+			if (i < (BT_CSIP_SIH_PRAND_SIZE - 1) && dest[i] != 0xFF) {
+				has_zeroes = true;
+			} else if ((dest[i] & 0x3F) != 0x3F) {
+				has_zeroes = true;
+			}
+			if (has_ones && has_zeroes) {
+				valid = true;
+				break;
+			}
```

Comment:
I understand the simplicity and the neatness, however, we are then back to using 4-byte variables for a 24-bit number for this validation, this is something I really wanted to avoid. I understand that this manual iteration doesn't look as neat, but I do think it gives meaning.